hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/hudi/HoodieSparkSqlWriter.scala

```diff
@@ -112,12 +112,15 @@ private[hudi] object HoodieSparkSqlWriter {
         val archiveLogFolder = parameters.getOrElse(
           HoodieTableConfig.HOODIE_ARCHIVELOG_FOLDER_PROP_NAME, "archived")
 
+        val partitionColumns = parameters.getOrElse(DataSourceWriteOptions.PARTITIONPATH_FIELD_OPT_KEY, null)
```

Comment:
the partition path can also be generated from a key generator, right? we need to think this case through more.

Reply:
Hi @vinothchandar , we need get the partition schema and pass it to the `HadoopFsRelation`,like this:

HadoopFsRelation(
        fileIndex,
        fileIndex.partitionSchema,
        fileIndex.dataSchema,
        bucketSpec = None,
        fileFormat = new ParquetFileFormat,
        optParams)(sqlContext.sparkSession)
By the generator, it is hard to construct the partition schema. So we need the partition columns.

Reply:
Understood. but what about a custom key generator which does not depend on `DataSourceWriteOptions.PARTITIONPATH_FIELD_OPT_KEY` ? Should we add handle some errors in that scenario? or turn off the use of file index?